### DISCUSSION TITLE
Updated Intel e1000e driver

### DISCUSSION THREAD

**Discussion by vanh20** [Security-Onion-Solutions/securityonion#6788] [Unsupported Versions] [answered]
I have seen [this](https://github.com/Security-Onion-Solutions/securityonion/discussions/6179) thread and I know what the answer will be from the developers. Yes, it's a fair reply. However, I'm appealing to the community this time. I've been at it since days before New Year and it has taken me into endless rabbit holes and hardware solutions which did not work.

We know this is a CentOS 7 (specific) thing. It's a fact of life which, I wish, did not exist. But yes... it does. Things keep breaking and things are so erratic that one can't predict what's going to happen next. I've got a main board that supports I219-V intel lan chip. It's my bad. Poor planning. Wrong decision. However, there must be a way by which you guys have updated your drivers that worked for SecurityOnion 2.3.91. What I have tried are as follows..

1. install the normal way (ISO burned on USB thumb) -> my usb to Ethernet adapter got recognized instantly.. no issues whatsoever at any stage. My built-in lan is identified by lspci, but not installed... it is a driver issue.
2. booted normally but only to setup networking (using my USB to LAN adapter) so I may be able to update.. then cancelled it to command prompt.
3. got the driver loaded on USB thumb and transferred it to the New Installation.
4. Ran sudo yum install <driverfile.rpm> , and it was a success... all downloaded perfectly... including a new kernel version (lower than the one from the ISO image)
5. rebooted so changes might be reflected and who knows what else need be. (I've been to so many trial and errors just to come this far)
6. Again, the installation script from SecurityOnion kicks in and asking me to install or to setup networking. Selected install.
7. It ran... until it encountered Running First Salt Checkin at 16%. I retried over and over and got to different stages every time, but there's always error somewhere. I have even followed a thread that basically says... that's a common problem with network install. And I am at my ropes end.

So, is there a way for the driver to be updated and still have a sweet ending? Or... any possible workaround??  If not, then I will just have to use two USB ethernet adapters for now. Gets recognized instantly. And Yes, I know that once the setup completes... everything gets version locked.

Thank you guys in advance.

**Comment by vanh20** [comment]
update:

I realized IP addresses gets blocked by my firewall via Suricata. I disabled the rule category for those alerts and did the installation again, this time with two USB LAN adapters. After a while when I came back, the installation was still unsuccessful giving multiple errors (from /root/sosetup.log) "Could not resolve host: repo.securityonion.net; Unknown error". Again, this is network related.

Question... is it not possible to have it all installed OFFLINE? What's the procedure?

TIA

**Comment by vanh20** [comment]
update:

I tried to do an airgap install. It was a success! Now, nothing gets into my SOC through the network.. I hope :D   I'm happy! The next thing is, how to do updates. 

Question:
    Can I pivot from this airgap install to the normal install from where I am now?

TIA

**Comment by vanh20** [comment]
update:

Updating is quite easy. [This](https://github.com/Security-Onion-Solutions/securityonion/discussions/6092) is confirmed.
1. do a so-rule-update on your airgap install. see [where](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/soup#L719) the file is coming from.
2. once inside the folder, [replace](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/common/tools/sbin/soup#L719) that file with the updated one from the signature maintainer.
3. see how many rules have been updated etc. etc. etc.


EDIT:
   I need to do the stuff above because I failed to read the documents. It sometimes can be so overwhelming. [This](https://docs.securityonion.net/en/2.3/airgap.html?highlight=airgap%20install) will give newbies the info we need about airgap install.

The only thing that needs to be resolved is how to pivot from airgap to live internet updating.. without a reinstall.

Anyway, this is my workaround, for now.

Thanks everyone!

**Comment by vanh20** [comment, marked as answer]
Final Update and Workaround (for me):

I tried to combine the first method of installing the driver above, and also installed all my favorite packages to help me monitor my installation. Then I rebooted to have the installation kick-in -- so I did an airgap install. There was error because IP address is detected over my LAN (e1000e built-in). I removed the cable for a while (eno1) and had it restarted again. One must keep the cable on the management interface, which I preferred to have a STATIC config and not a DHCP one. Then this time the airgap install went and completed.

There was error on the installation and upon seeing /root/sosetup.log it says:

_**sudo cat /root/sosetup.log | grep error**_

‘/root/SecurityOnion/agrules/sigma/rules/application/app_sqlinjection_errors.yml’ -> ‘/nsm/repo/rules/sigma/rules/application/app_sqlinjection_errors.yml’
‘/root/SecurityOnion/agrules/sigma/rules/web/web_apache_threading_error.yml’ -> ‘/nsm/repo/rules/sigma/rules/web/web_apache_threading_error.yml’
*** Restarting Salt to fix any SSL errors. ***
+log_level: error
+log_level_logfile: error
+log_level: error
+log_level_logfile: error
error: HTTP 404: Not Found
error: HTTP 404: Not Found

I did the following:

1. searched`` the forum for ‘/nsm/repo/rules/sigma/rules/application/app_sqlinjection_errors.yml’ and /nsm/repo/rules/sigma/rules/web/web_apache_threading_error.yml", found none.` NOTE: the files exists on their appropriate locations.
2. reboot
3. ran _**sudo so-status**_ -> all green [ok]
4. **_sudo salt-call state.highstate_**, **_sudo so-salt-stop_**, and **_sudo so-salt-start_** commands yielded no errors
5. tested to download an eicar pdf file, strelka got it at /nsm/strelka/processed/
6. used hunt and alert... the file download is indeed reflected and can even see the captured pcap.

All seem to be well and working??

If it does work for you, please let me know. But if the log above doesn't seem to be a resolved issue, then please let me know with the possible solution if it's not asking for too much. :)

Again, if there's a way to pivot from airgap to normal install, please let me know as well.

TIA!

**Comment by bonjoji** [comment]
- Install OS
- make sure networking works
- Then install needed packages and updates
- Enable firewall
- Check everything allowed works
- Install ids/ips
- Configure, enable, activate, test everything is working
- Have a coffee break lol

**Reply by vanh20** [reply]
Hello bonjoji! 

   You are absolutely correct in your analysis and troubleshooting steps! Thanks man! Yes! Going through rabbit holes makes one so tired and miss so many things! :D Yes I followed that path. Firewall is not preventing any and I have disabled rules in Suricata. I also have checked the alerts and all seem fine. No more IP addresses are being blocked. Anyway, I will still try to see which IP addresses might have been blocked, though I know I've tried to clear Snort2c from the Tables of my pfSense FW. But as of now, My setup is working fine offline/airgap. And I think I will keep it this way for now until the Omicron surge subsides.

   When I have the time later, I will try again from zero and see if I can have an error free normal installation. But I kinda liked the airgap setup this time... makes it hard for my SOC to be taken over by the nice guys around here. :D

   Thanks for the advice! :)
